lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -159,6 +168,7 @@ def get_collection_dependencies(self, collection_candidate):
                 get_direct_collection_dependencies
             )(collection_candidate)
 
+        self._offline_is_unsupported()
```

Comment:
Why don't we put this at the beginning of the `get_collection_version_metadata()` method so that it'd be an entry-check, like in other places?

Reply:
I considered changing `get_collection_version_metadata` to allow retrieving metadata from concrete artifacts, but figured if that was useful it could be done as a follow up.

This is in the middle of `get_collection_dependencies` because we do get dependencies from concrete artifacts without GalaxyAPI calls. There's a return statement above this.

Reply:
I think you meant this was unnecessary, rather than it should be moved (since get_collection_version_metadata() also checks). Fixed that.